core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java

```diff
@@ -123,11 +128,21 @@ public void authenticate(String connectionUser, String impersonationUser)
           }
         }
       } catch (IOException e) {
-        throw new AuthenticationException(ExceptionMessage.IMPERSONATION_GROUPS_FAILED
-            .getMessage(impersonationUser, connectionUser), e);
+        throw new AuthenticationException(String.format(
+            "Failed to authenticate client user=\"%s\" connecting to Alluxio master and "
+                + "impersonating as impersonationUser=\"%s\" to access Alluxio file system. "
+                + "Failed to get the allowed impersonation groups for impersonationUser=\"%s\"."
```

Comment:
This is not the actual issue here. The issue is that alluxio cannot retrieve the groups that the impersonation user is a part of. It is not actually related to any impersonation configuration.